analyzers/src/SonarAnalyzer.CSharp/Rules/ReturnEmptyCollectionInsteadOfNull.cs

```diff
@@ -23,131 +23,119 @@ namespace SonarAnalyzer.Rules.CSharp
+                var nullOrDefaultLiterals = GetReturnNullOrDefaultExpressions(body)
                     .Select(returnStatement => returnStatement.GetLocation())
                     .ToList();
-                if (returnNullStatements.Count > 0 &&
-                    IsReturningCollection(context))
+
+                if (nullOrDefaultLiterals.Count > 0 && IsReturningCollection(context))
                 {
-                    context.ReportIssue(rule.CreateDiagnostic(context.Compilation, returnNullStatements[0], additionalLocations: returnNullStatements.Skip(1)));
+                    context.ReportIssue(Rule.CreateDiagnostic(context.Compilation, nullOrDefaultLiterals[0], additionalLocations: nullOrDefaultLiterals.Skip(1)));
                 }
             }
         }
 
         private static bool IsReturningCollection(SyntaxNodeAnalysisContext context)
         {
             var symbol = context.SemanticModel.GetDeclaredSymbol(context.Node);
-
             var methodSymbol = (symbol as IPropertySymbol)?.GetMethod ?? symbol as IMethodSymbol;
 
-            return methodSymbol != null &&
-                !methodSymbol.ReturnType.Is(KnownType.System_String) &&
-                !methodSymbol.ReturnType.DerivesFrom(KnownType.System_Xml_XmlNode) &&
-                methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
+            return methodSymbol != null
+                && !methodSymbol.ReturnType.Is(KnownType.System_String)
+                && !methodSymbol.ReturnType.DerivesFrom(KnownType.System_Xml_XmlNode)
+                && methodSymbol.ReturnType.DerivesOrImplementsAny(CollectionTypes);
         }
 
-        private static ArrowExpressionClauseSyntax GetExpressionBody(SyntaxNode node)
-        {
-            switch (node.Kind())
+        private static ArrowExpressionClauseSyntax GetExpressionBody(SyntaxNode node) =>
+            node switch
             {
-                case SyntaxKind.MethodDeclaration:
-                    return ((MethodDeclarationSyntax)node).ExpressionBody;
-
-                case SyntaxKind.PropertyDeclaration:
-                    var property = (PropertyDeclarationSyntax)node;
-
-                    if (property.ExpressionBody != null)
-                    {
-                        return property.ExpressionBody;
-                    }
-
-                    return property.AccessorList?.Accessors
-                        .FirstOrDefault(accessor => accessor.IsKind(SyntaxKind.GetAccessorDeclaration))
-                        ?.ExpressionBody();
+                MethodDeclarationSyntax method => method.ExpressionBody,
+                PropertyDeclarationSyntax property => property.ExpressionBody ?? GetGetAccessor(property)?.ExpressionBody(),
+                var _ when LocalFunctionStatementSyntaxWrapper.IsInstance(node) => ((LocalFunctionStatementSyntaxWrapper)node).ExpressionBody,
+                OperatorDeclarationSyntax @operator => @operator.ExpressionBody,
```

Comment:
What about indexer:
```cs
public IEnumerable<string> this[int i] => null;
```
I'm not sure, but are conversion operators covered by this? I think it is a different syntax kind ConversionOperatorDeclaration):
```cs
public static implicit operator IEnumerable<int>(C c) => null;
```
```

Reply:
That is outside of the scope of this PR. Just like with S2190 in [this PR](https://github.com/SonarSource/sonar-dotnet/pull/6435) the focus was on Operators as we expect more usage of those with C#11. There should be a new issue opened with reproducers for those SyntaxKinds you have mentioned..